src/BootKeyboard/BootKeyboard.h

```diff
@@ -78,6 +78,7 @@ class BootKeyboard_ : public PluggableUSBModule {
 
   uint8_t epType[1];
   uint8_t protocol;
+  bool boot_protocol_{false};
```

Comment:
`false` => boot protocol
`true` => report (NKRO) protocol

Reply:
This adds a state variable whose only purpose is to track changes requested by Kaleidoscope, and to feed them back to Kaleidoscope via `getProtocol`. I'm not sure it belongs here. On the other hand, `setProtocol` is of dubious value as well, because (before these changes) it can force the interface to answer `GET_PROTOCOL` with a different protocol than the host previously selected, which might cause the host to decide that the interface is misbehaving.